api/views/yangSearch/grep_search.py

```diff
@@ -187,15 +212,27 @@ def _get_modules_from_cursor(
         except IndexError:
             return None
 
-    def _get_all_modules_with_filename_extension(self) -> list[str]:
-        all_modules = cache.get(self.listdir_results_cache_key)
+    def _get_all_modules_with_filename_extension(self, return_set: bool = False) -> t.Union[set[str], list[str]]:
```

Comment:
I don't think we should complicate the return type like this. It confuses the type checker and the performance impact for the conversion is negligible (less than a ms for 10000 64 char strings on my machine).